java/code/src/com/suse/manager/webui/menu/MenuTree.java

```diff
@@ -71,282 +71,354 @@ public  MenuTree(AclFactory aclFactoryIn) {
         MenuItemList nodes = new MenuItemList();
 
         if (checkAcl(user, "user_authenticated()")) {
-            // Home
-            nodes.add(new MenuItem("Home").withIcon("fa-home")
-                .addChild(new MenuItem("Overview").withPrimaryUrl("/rhn/YourRhn.do"))
-                .addChild(new MenuItem("Notification Messages").withPrimaryUrl("/rhn/manager/notification-messages"))
-                .addChild(new MenuItem("User Account")
+            nodes.add(getHomeNode(adminRoles));
+            nodes.add(getSystemsNode(user, adminRoles));
+
+            // Clusters
+            nodes.add(new MenuItem("clusters.nav.title").withIcon("spacewalk-icon-clusters")
+                    .addChild(new MenuItem("clusters.nav.overview").withPrimaryUrl("/rhn/manager/clusters")
+                        .withDir("/rhn/manager/cluster"))
+                    .addChild(new MenuItem("clusters.nav.add").withPrimaryUrl("/rhn/manager/clusters/add")
+                            .withVisibility(adminRoles.get("clusters"))));
+
+            // Salt
+            nodes.add(new MenuItem("Salt").withIcon("spacewalk-icon-salt")
+                .addChild(new MenuItem("Keys").withPrimaryUrl("/rhn/manager/systems/keys"))
+                .addChild(new MenuItem("Remote Commands").withPrimaryUrl("/rhn/manager/systems/cmd"))
+                .addChild(new MenuItem("Formula Catalog").withPrimaryUrl("/rhn/manager/formula-catalog")
+                    .withVisibility(adminRoles.get("org"))));
```

Comment:
Why you did not extract the `Clusters` and `Salt` pieces as well? It would be very nice and readable to have just the first Menu level as a plain list of submethods as it follows after these lines.

Reply:
that was laziness and those are super small... but I agree those could be extracted as well for coherence.

Reply:
done now